crates/engine/tree/src/tree/mod.rs

```diff
@@ -253,7 +264,7 @@ impl TreeState {
 
         while old_hash != current_hash {
             if let Some(block) = self.blocks_by_hash.get(&old_hash) {
-                old_chain.push(block.clone());
+                old_chain.insert(0, block.clone());
```

Comment:
since we seem to be only inserting at the front, it might be worth doing `push` and then `reverse` like we do with `new_chain`

Reply:
makes sense, done here https://github.com/paradigmxyz/reth/pull/10320/commits/3f8a5bc62fb90be0ebf32dc89659a9b1e1392129